packages/eas-cli/src/credentials/context.ts

```diff
@@ -27,6 +28,7 @@ export class CredentialsContext {
       nonInteractive?: boolean;
       projectDir: string;
       user: Actor;
+      buildProfile?: BuildProfile;
```

Comment:
nit pick: build profile concept should not be mixed with credential internally if it's not necessary, you could pass here env object, or maybe entire exp object